lib/logstash/inputs/jmx.rb

```diff
@@ -189,7 +189,11 @@ def send_event_to_queue(queue,host,metric_path,metric_value)
     else
       @logger.debug("The value #{metric_value} is not of type number: #{metric_value.class}")
       event.set('metric_path', metric_path_substituted)
-      event.set('metric_value_string', metric_value.to_s)
+      if metric_value.respond_to?(:java_object) && metric_value.java_object.class == Java::JavaArray
+        event.set('metric_value_string', metric_value.to_a.to_s)
+      else
+        event.set('metric_value_string', metric_value.to_s)
+      end
     end
```

Comment:
I think we could simplify the code by doing this `Array(metric_value).to_s`, if the source is already an array it wont get converted and this will remove the conditionals.

So the code would become this

```ruby
event.set('metric_value_string', Array(metric_value).to_s)
```